### PR TITLE
hold unhold changes for beta 3

### PR DESCRIPTION
**Pull request by BalajiUmmadisetti**
# Description
US : https://skype.visualstudio.com/SPOOL/_workitems/edit/3621291
while testing SDK getting revapi - java.method.numberOfParametersChanged issue, so didn't change existing constructor , just added new constructor method with 3 params.
Added CallParticipant constructor with 3 params
public CallParticipant(CommunicationIdentifier identifier, Boolean isMuted, Boolean isOnHold).

Tested using Contoso Bank Outbound call quick start.
Tested Scenarios:
1.hold
2.holdWithResponse
3.unhold
4.unholdWithResponse
5.CallParticipant with isOnHold option.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+     * Operation context.
+     */
+    private String operationContext;
+
+    /**
+     * Create a new HoldOptions object.
+     * @param targetParticipant Participant to be put on hold.
+     * @param playSourceInfo Audio to be played while on hold.
+     */
+    public HoldOptions(CommunicationIdentifier targetParticipant, PlaySource playSourceInfo) {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java:33]: `playSourceInfo` is an optional parameter. Please, remove from the constructor and add a setter for it.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java [hunk 2]
```diff
@@ -378,6 +379,32 @@ private CallMediaAsync getMockCallMedia(int expectedStatusCode) {
...
         return callConnection.getCallMediaAsync();
     }
+    
+    @Test
+    public void holdWithResponseTest() {
+
+        callMedia = getMockCallMedia(200);
+        HoldOptions options = new HoldOptions(
+            new CommunicationUserIdentifier("id"),
+            new TextSource().setText("audio to play"));
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java:389]: Please, add a test with no prompt.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java [hunk 3]
```diff
@@ -186,6 +186,33 @@ public void sendDtmfTonesWithResponseTest() {
...
+    @Test
+    public void holdWithResponseTest() {
+        CallConnection callConnection =
+            CallAutomationUnitTestBase.getCallConnection(new ArrayList<>(
+                Collections.singletonList(new AbstractMap.SimpleEntry<>("", 200)))
+            );
+        callMedia = callConnection.getCallMedia();
+        HoldOptions options = new HoldOptions(
+            new CommunicationUserIdentifier("id"),
+            new TextSource().setText("audio to play"));
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java:199]: Please, add a test with no prompt.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+// Licensed under the MIT License.
+
+package com.azure.communication.callautomation.models;
+
+import com.azure.communication.common.CommunicationIdentifier;
+
+/**
+ * Options for the Hold operation.
+ */
+public class HoldOptions {
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java:11]: this is missing the optional call back URI.

Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java:11]: added optional call back URI

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java (hunk 1 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/HoldOptions.java]: made playSourceInfo as optional parameter.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java (hunk 3 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaUnitTests.java]: added a test with no prompt.

File: sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java (hunk 2 above)
Comment (BalajiUmmadisetti) [reply sdk/communication/azure-communication-callautomation/src/test/java/com/azure/communication/callautomation/CallMediaAsyncUnitTests.java]: added a test with no prompt.

**Comment by cochi2** [comment]
As in [here](https://github.com/Azure/azure-sdk-for-java/pull/39085), this PR is missing the changes to `CallParticipant` and changes to the EventParser and add the `HoldFailed` event to the Models/Events folder.

**Review by Paresh-Arvind-Patil — APPROVED** [review]

**Pull request merged**